src/main/java/gregtech/api/recipes/RecipeBuilder.java

```diff
@@ -886,19 +898,26 @@ protected static String getRequiredString(int max, int actual, @NotNull String t
         return out;
     }
 
-    protected R onBuild(Consumer<R> consumer) {
-        this.onBuildAction = consumer;
+    protected R onBuild(@NotNull List<@NotNull RecipeBuildAction<R>> actions) {
```

Comment:
Should we also have a builder parameter for `onBuild` that accepts a single RecipeBuildAction, and then adds it to the list of actions (if the list exists), or creates the list if it does not exist?

Also, if it is required to pass in the List of all RecipeBuildActions here, we should probably add a getter for it, so that the field does not have to be called directly.